pysat/_custom.py

```diff
@@ -126,7 +126,8 @@ def _apply_all(self, sat):
         if len(self._functions) > 0:
             for func, arg, kwarg, kind in zip(self._functions, self._args,
                                               self._kwargs, self._kind):
-                if len(sat.data) > 0:
+                ind = sat.data.index if sat.pandas_format else sat.data.indexes
```

Comment:
I think this should be if not sat.empty:

Reply:
I agree, but we need to create a routine that adds an `empty` property to the xr.Dataset.

Reply:
I don't think so, that is the point of the 'empty' attribute attached to each pysat.Instrument object. It internally deals with the differences between pandas and xarray.

Reply:
You're right.  It's the `_instrument` line I'm thinking of.

Reply:
fixed here